velox/dwio/common/Options.cpp

```diff
@@ -77,4 +78,85 @@ ColumnReaderOptions makeColumnReaderOptions(const ReaderOptions& options) {
   return columnReaderOptions;
 }
 
+folly::dynamic WriterOptions::serialize() const {
+  folly::dynamic obj = folly::dynamic::object;
+  obj["name"] = "WriterOptions";
+
+  // 1) Schema
+  if (schema) {
+    obj["schema"] = schema->serialize();
+  }
+
+  // 2) compressionKind
+  if (compressionKind) {
+    obj["compressionKind"] = static_cast<int>(*compressionKind);
+  }
+
+  // 3) serdeParameters
+  if (!serdeParameters.empty()) {
+    folly::dynamic mapObj = folly::dynamic::object;
+    for (auto& [k, v] : serdeParameters) {
+      mapObj[k] = v;
+    }
+    obj["serdeParameters"] = std::move(mapObj);
+  }
+
+  // 4) sessionTimezoneName
+  if (!sessionTimezoneName.empty()) {
+    obj["sessionTimezoneName"] = sessionTimezoneName;
+  }
+
+  // 5) adjustTimestampToTimezone
+  obj["adjustTimestampToTimezone"] = adjustTimestampToTimezone;
+
+  // We do *not* serialize pool, spillConfig, nonReclaimableSection,
```

Comment:
Would you move this comment to the header file? I assume the caller of 'create' would need to be aware of these.